src/puppetlabs/pcp/client.clj

```diff
@@ -257,27 +278,34 @@
   (let [{:keys [cert private-key cacert]} ssl-files]
     (ssl-utils/pems->ssl-context cert private-key cacert)))
 
-;; private helpers for the ssl/websockets setup
-(s/defn ^:private make-ssl-context :- SslContextFactory
-  "Returns an SslContextFactory that does client authentication based on the
-  client certificate named"
+;; public helpers for the ssl/websockets setup, also used by pcp-broker's test client
+(s/defn ^:public make-ssl-context :- SslContextFactory$Client
+  "Returns an SslContextFactory$Client that does client authentication based on the
+  client certificate named."
   [params]
-  (let [factory (SslContextFactory.)
+  (let [factory (SslContextFactory$Client.)
         ssl-context (get-ssl-context (:ssl-context params))]
     (.setSslContext factory ssl-context)
-    (.setNeedClientAuth factory true)
+    ;;(.setNeedClientAuth factory true) ;; not available with $Client, problem?
```

Comment:
Is there a logical replacement, or a way to ensure that auth is done?

Reply:
Does SSL need client authentication? Looking into more what this is. Seems kind of like the right idea to look at the Jetty code as it might be the best documentation they have: https://github.com/jetty/jetty.project/blob/a9878863aa59bf953e35dc37d668ff576e28065f/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java#L2230

Reply:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/javax/net/ssl/SSLEngine.html#setNeedClientAuth(boolean) since it does exist on SslContextFactory$Server it seems it can just be set there in tk-jetty-10.